File: Models/Code.cs
using System;
using System.Collections.Generic;
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Newtonsoft.Json;

namespace AccountAPI.Models
{
	public class Code
    {
        [Key]
        public int CodeId {get;set;}
        [Required]
        public string CodeString {get;set;}
        [Required]
        public bool UsedStatus {get;set;}
        public DateTime CreatedAt {get;set;} = DateTime.Now;
        public DateTime UpdatedAt {get;set;} = DateTime.Now;

        // Sets accounts through composite key
        public int? EmailAccountId {get;set;}
        public int? PlatformId {get;set;}
        public Account Account {get;set;}

        // Sets Game though foreign key
        [Required]
        public int GameId {get;set;}
        public Game Game {get;set;}
    }
}

File: Client/src/app/Models/code.ts
export interface CodeList {
  Id: number;
  Code: string;
  Used: boolean;
  Account: string;
  Game: string;
}

export interface CodeAdd {
  CodeString: string,
  UsedStatus: boolean;
  EmailAccountId: null;
  PlatformId: number;
  GameId: number;
}

export interface CodeEdit {
  CodeId: number;
  CodeString: string;
  UsedStatus: boolean;
  EmailAccountId: number;
  PlatformId: number;
  AccountId: number;
  // Account: string;
  GameId: number;
}

export interface Code {
  CodeId: number;
  CodeString: string;
  AccountId: number;
  Account: string;
  UsedStatus: boolean;
  EmailAccountId: number;
  PlatformId: number;
  GameId: number;
  Game: string;
}
